lib/utils/ec2.go

```diff
@@ -109,8 +111,21 @@ func NewInstanceMetadataClient(ctx context.Context) (*InstanceMetadataClient, er
 
 // IsAvailable checks if instance metadata is available.
 func (client *InstanceMetadataClient) IsAvailable(ctx context.Context) bool {
-	_, err := client.getMetadata(ctx, "")
-	return err == nil
+	// Doing this check via imds.Client.GetMetadata() involves several unrelated requests and takes a few seconds
+	// to complete when not on EC2. This approach is faster.
+	httpClient := http.Client{
+		Timeout: 250 * time.Millisecond,
```

Comment:
Is 250ms enough for metadata endpoint?

Reply:
In my testing, the whole check takes about 3ms when on EC2, but there's no harm in extending the timeout (other than the check taking longer) if that's what we want.